modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java

```diff
@@ -380,6 +411,33 @@ private DepotEntry _addOrUpdateDepotEntry(
 			_updateDLSizeLimitConfiguration(
 				assetLibrary, group.getGroupId(), mimeTypeSizeLimits);
 
+			List<DepotEntryGroupRel> depotEntryGroupRels =
+				_depotEntryGroupRelService.getDepotEntryGroupRels(
+					depotEntry, QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+			List<Long> groupIds = _getEligibleGroupIds(
```

Comment:
This whole logic seems duplicated, can you please reuse the code for what is possible?